src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -1441,6 +1479,24 @@ public String getTriggerConfigURL() {
      */
     @DataBoundSetter
     public void setTriggerConfigURL(String triggerConfigURL) {
+        // If the trigger URL has changed from the previous value, then we can't use
+        // the project list that we might already have loaded from the last time this trigger
+        // was started.  Just make a note of that here; we'll deal with it in "start()".
+        if (triggerConfigURL == null) {
+            if (this.triggerConfigURL != null) {
+                // We need to start fresh when the URL has changed.
+                this.dynamicConfigurationChangedFromLastStart = true;
+            }
+        } else {
+            if (!triggerConfigURL.equals(this.triggerConfigURL)) {
+                // We need to start fresh when the URL has changed.
+                this.dynamicConfigurationChangedFromLastStart = true;
```

Comment:
Have you tested that this actually works? IIRC when configuring the trigger a new instance is created, both via the UI and via pipeline script, so it's not the same instance that gets the `stop`  and `start` calls.

Reply:
No, I haven't begun testing yet.  Once you're happy with the approach (and looks like you are), I'll actually deploy this to my test servers and work through my test cases.

Reply:
I had always assumed that it was the same instance.  If you have documented proof of that fact, then we can probably trim some of the new code a bit.

(I've struggled with the lack of up-to-date Jenkins plugin docs; mostly I've just been piecing together how things work from looking at existing plugins and hoping that they're right.)

Reply:
https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/AbstractProject.java#L1804-L1808

Reply:
Did you resolve this?

Reply:
I don't think that I ever saw this comment (sorry).  Is that a confirmation that a trigger instance will _never_ be started again after it has been stopped?  If so, I'll remove all of the restart-related code.

Reply:
`dynamicConfigurationChangedFromLastStart` has been completely removed.

Reply:
yepp, that link confirms that a reconfiguration of a job will create a new trigger object.